rskj-core/src/test/java/co/rsk/rpc/netty/Web3WebSocketServerTest.java

```diff
@@ -96,6 +100,358 @@ void tearDown() {
+                maxAggregatedFrameSize
+        );
+        websocketServer.start();
+
+        OkHttpClient wsClient = new OkHttpClient();
+        Request wsRequest = new Request.Builder().url("ws://localhost:" + randomPort + serverPath).build();
+        WebSocketCall wsCall = WebSocketCall.create(wsClient, wsRequest);
+
+        CountDownLatch wsAsyncResultLatch = new CountDownLatch(1);
+        CountDownLatch wsAsyncCloseLatch = new CountDownLatch(1);
+        AtomicReference<Exception> failureReference = new AtomicReference<>();
+        wsCall.enqueue(new WebSocketListener() {
+
+            private WebSocket webSocket;
+
+            @Override
+            public void onOpen(WebSocket webSocket, Response response) {
+                wsExecutor.submit(() -> {
+                    RequestBody body = RequestBody.create(WebSocket.TEXT, msg);
+                    try {
+                        this.webSocket = webSocket;
+                        this.webSocket.sendMessage(body);
+                        this.webSocket.close(1000, null);
+                    } catch (IOException e) {
+                        failureReference.set(e);
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(IOException e, Response response) {
+                failureReference.set(e);
+            }
+
+            @Override
+            public void onMessage(ResponseBody message) throws IOException {
+                JsonNode jsonRpcResponse = JacksonParserUtil.readTree(OBJECT_MAPPER, message.bytes());
+
+                Assertions.assertEquals(jsonRpcResponse.get("error").get("code").asInt(), ErrorResolver.JsonError.INVALID_REQUEST.code);
+                Assertions.assertEquals("Cannot dispatch batch requests. 1 is the max number of supported batch requests", jsonRpcResponse.get("error").get("message").asText());
+
+                message.close();
+                wsAsyncResultLatch.countDown();
+            }
+
+            @Override
+            public void onPong(Buffer payload) {
+            }
+
+            @Override
+            public void onClose(int code, String reason) {
+                wsAsyncCloseLatch.countDown();
+            }
+        });
+
+        if (!wsAsyncResultLatch.await(10, TimeUnit.SECONDS)) {
+            fail("Result timed out");
+        }
+
+        if (!wsAsyncCloseLatch.await(10, TimeUnit.SECONDS)) {
+            fail("Close timed out");
+        }
+
+        websocketServer.stop();
+
+        Exception failure = failureReference.get();
+        if (failure != null) {
+            failure.printStackTrace();
+            fail(failure.getMessage());
+        }
+    }
+
+    @Test
+    void testStackOverflowErrorInRequest() throws Exception {
+        List<String> messages = new ArrayList<>();
+        String content = this.getJsonRpcDummyMessageStr("value");
+
+        for (long i = 0; i < 999_999; i++) {
+            content = String.format("[%s]", content);
```

Comment:
I believe `String.format` is highly inefficient for such big number and can degradete test execution time significantly. I'd recommend using `StringBuilder` here with initial capacity. Btw, how much time does this test take for you?

Reply:
Fixed this